libraries/Microsoft.Bot.Builder.LanguageGeneration/Extensions.cs

```diff
@@ -79,6 +79,23 @@ public static string Escape(this string text)
             }));
         }
 
+        /// <summary>
+        /// trim expression. @{abc} => abc,  @{a == {}} => a == {}.
+        /// </summary>
+        /// <param name="expression">input expression string.</param>
+        /// <returns>pure expression string.</returns>
+        public static string TrimExpression(this string expression)
+        {
+            var result = expression.Trim().TrimStart('@').Trim();
+
+            if (result.StartsWith("{") && result.EndsWith("}"))
+            {
+                result = result.Substring(1, result.Length - 2);
+            }
+
+            return result;
+        }
```

Comment:
I don't think this is appropriate any more with delta I made to do StringExpression, etc. 
Can we talk with vishwac about this?

Reply:
> I don't think this is appropriate any more with delta I made to do StringExpression, etc.
> Can we talk with vishwac about this?

This is the context of LG, we would handle it at the bottom layer of dialog parsing. It may have nothing to do with the adaptive layer.   @vishwacsena

Reply:
This is just a very low level helper method to trim string in LG context like
```
# hello
- @{user.name} // this will get trimmed to user.name, then send to expression engine to evaluate 
``` 

It's looks like doing very similar trimming thing, but actually very related to stringExpression, which i suppose it's to handle property assignment in AdaptiveDialog level.

From a layering perspective,
* this is a low level LG helper method working in the parsing phrase,  only aware of how expression may present in LG text, and things like "=xxx" don't apply here, it is a strict expression. 
* while "stringExpression" is a higher level concept in property level, defining what to be treated as a "expression" when parsing dialog json.

Reply:
No, it doesn't.  The expression parser should know about @{ }, that's a function above that, no? 
Are you saying this makes sense?
"@{substring(@{user.name}, 15)}"
?

If we are doing string interpolation, then it's a function which is on top of expression evaluation which injects expression results into a string. 
"Hello @{substring(user.name, 1)}!!!"
should extract 
    substring(user.name, 1) => insert into the outer string the result as a string
"Hello omas"
I don't think @ { } syntax has anything to do with expression evaluation, only as a mechanism for defining the expression to evaluate and the place to put the result.

Am I missing something?

Reply:
Offline chatted with @tomlm, and i think Tom gets the intention, to be crystal clear, this is not a function in expression parser, it's a helper in LG parser. 

From the layer perspective,
* ExpressionParser, the bottom layer, knows nothing about "@{}" notion *today* (before we support string interpolation directly in Expression),  it's just a notion upper layer used to define (LG or Dialog or TypeLoader) which part is expression.  So when we call parser to parse, we don't pass in string with @{}
```C#
var expStr = "user.name"; // this don't contains @{}
var exp = new ExpressionEngine().parse(expStr)
```
* LGParser will know about @{}, when it sees things like @{user.name}, it will trim @{} and feed into expression parser, which is what this helper does

And yes, when we support string interpolation in Expression (which is in another PR), we will move some LG capability into Expression, which perhaps can include this helper, because LG is essentially one "string interpolation" approach.